bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ItemRegistry.java

```diff
@@ -25,6 +25,14 @@
 public interface ItemRegistry extends Registry<Item, String> {
 
     /**
+     * This methods checks if an item is part of the registry.
+     *
+     * @param name the item name
+     * @return true if there is an item with given name, otherwise false
+     */
+    public boolean containsItem(String name);
```

Comment:
The registry will hold (for the ItemRegistry) elements of type Item, so we cannot rely on "name" (an item attribute) in the Regitry class.
We could use getItem(name) and catch ItemNotFoundExcpetion. I am not a fan of an exception based control flow, if you like that. Okay